### DISCUSSION TITLE
TG-FileStreamBot 3.0.0 is out

### DISCUSSION THREAD

**Discussion by EverythingSuckz** [EverythingSuckz/TG-FileStreamBot#136] [Announcements]
☁️ [File stream bot version 3.0.0](https://github.com/EverythingSuckz/TG-FileStreamBot/releases/tag/3.0.0) is out

🆕 What's new?

- 👤 Support for allowed users
   Now you can restrict other users from using the bot by setting your user ID to `ALLOWED_USERS` config. You may also add multiple IDs by separating each ID using commas (`,`) 

- 🌐 Auto detect and use public IP
The bot now auto detects and uses public IP for links if they are available

- 👥  Fix bot replying in groups
   Earlier, the bot used to reply to `/start` commands in groups and now that's fixed.

- 🔗 New text formatting to links
 Check that out yourself

- 🔧 Other minor bug fixes and features

Please report bugs in the [discussion chat](https://t.me/AlteredVoid) or in the [GitHub issues](https://github.com/EverythingSuckz/TG-FileStreamBot).

**Comment by 5bc2e7** [comment]
I use docker compose up -d to run the project, I can download the file normally but there are a lot of error messages: Error while copying stream

When I use the IDM downloader, I observed that single threaded downloads of the same file produce only one error, while multi-threaded downloads produce errors all the time But there is no change in download speed.

> fsb  | 14/02/2024 03:55 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:37538: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 15:55:25 | 200 |  7.817829855s |  155.94.xxx.xxx | GET      "/stream/9?hash=d2b5a4&d=true"
fsb  | [GIN] 2024/02/14 - 15:56:51 | 200 |         1m27s |  155.94.xxx.xxx | GET      "/stream/9?hash=d2b5a4&d=true"
fsb  | 14/02/2024 03:58 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:50152: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 15:58:19 | 200 |  1.053508177s |  155.94.xxx.xxx | GET      "/stream/10?hash=a7b65a"
fsb  | [GIN] 2024/02/14 - 15:59:29 | 200 |         1m10s |  155.94.xxx.xxx | GET      "/stream/10?hash=a7b65a"
fsb  | 14/02/2024 04:01 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:59766: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 16:01:38 | 200 |  1.775129468s |  155.94.xxx.xxx | GET      "/stream/11?hash=847d62"
fsb  | 14/02/2024 04:02 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:59776: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 16:02:15 | 200 | 37.043437544s |  155.94.xxx.xxx | GET      "/stream/11?hash=847d62"
fsb  | 14/02/2024 04:02 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:47192: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 16:02:29 | 200 |  2.734985657s |  155.94.xxx.xxx | GET      "/stream/11?hash=847d62&d=true"
fsb  | 14/02/2024 04:02 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:47194: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 16:02:43 | 200 | 15.515843996s |  155.94.xxx.xxx | GET      "/stream/11?hash=847d62&d=true"
fsb  | 14/02/2024 04:02 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:34282: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 16:02:45 | 200 |  1.878350015s |  155.94.xxx.xxx | GET      "/stream/11?hash=847d62"
fsb  | 14/02/2024 04:03 PM	INFO	routes.Stream	Content-Range	{"start": 238080, "end": 2030538891, "fileSize": 2030538892}
fsb  | 14/02/2024 04:03 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:43866: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 16:03:30 | 200 |  1.093829497s |  155.94.xxx.xxx | GET      "/stream/12?hash=3400fd&d=true"
fsb  | 14/02/2024 04:03 PM	INFO	routes.Stream	Content-Range	{"start": 1015394325, "end": 2030538891, "fileSize": 2030538892}
fsb  | 14/02/2024 04:03 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:43882: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 16:03:30 | 206 |  345.961525ms |  155.94.xxx.xxx | GET      "/stream/12?hash=3400fd&d=true"
fsb  | 14/02/2024 04:03 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:43886: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 16:03:31 | 206 |  489.361598ms |  155.94.xxx.xxx | GET      "/stream/12?hash=3400fd&d=true"
fsb  | 14/02/2024 04:03 PM	INFO	routes.Stream	Content-Range	{"start": 1015394325, "end": 2030538891, "fileSize": 2030538892}
fsb  | 14/02/2024 04:03 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:35184: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 16:03:36 | 206 |  350.205199ms |  155.94.xxx.xxx | GET      "/stream/12?hash=3400fd&d=true"
fsb  | 14/02/2024 04:03 PM	INFO	routes.Stream	Content-Range	{"start": 1015394325, "end": 2030538891, "fileSize": 2030538892}
fsb  | 14/02/2024 04:03 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:40980: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 16:03:46 | 206 |  343.131783ms |  155.94.xxx.xxx | GET      "/stream/12?hash=3400fd&d=true"
fsb  | 14/02/2024 04:04 PM	INFO	routes.Stream	Content-Range	{"start": 1015394325, "end": 2030538891, "fileSize": 2030538892}
fsb  | 14/02/2024 04:04 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:43986: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 16:04:00 | 206 |  519.271591ms |  155.94.xxx.xxx | GET      "/stream/12?hash=3400fd&d=true"
fsb  | 14/02/2024 04:04 PM	INFO	routes.Stream	Content-Range	{"start": 1015394325, "end": 2030538891, "fileSize": 2030538892}
fsb  | 14/02/2024 04:04 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:49816: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 16:04:18 | 206 |  322.599754ms |  155.94.xxx.xxx | GET      "/stream/12?hash=3400fd&d=true"
fsb  | 14/02/2024 04:04 PM	INFO	routes.Stream	Content-Range	{"start": 1015394325, "end": 2030538891, "fileSize": 2030538892}
fsb  | 14/02/2024 04:04 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:36948: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 16:04:40 | 206 |  663.384321ms |  155.94.xxx.xxx | GET      "/stream/12?hash=3400fd&d=true"
fsb  | 14/02/2024 04:05 PM	INFO	routes.Stream	Content-Range	{"start": 1015394325, "end": 2030538891, "fileSize": 2030538892}
fsb  | 14/02/2024 04:05 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:44596: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 16:05:06 | 206 |  374.157187ms |  155.94.xxx.xxx | GET      "/stream/12?hash=3400fd&d=true"
fsb  | 14/02/2024 04:05 PM	INFO	routes.Stream	Content-Range	{"start": 1522972448, "end": 2030538891, "fileSize": 2030538892}
fsb  | 14/02/2024 04:05 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:40908: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 16:05:31 | 206 |  409.013128ms |  155.94.xxx.xxx | GET      "/stream/12?hash=3400fd&d=true"
fsb  | 14/02/2024 04:06 PM	INFO	routes.Stream	Content-Range	{"start": 1522972448, "end": 2030538891, "fileSize": 2030538892}
fsb  | 14/02/2024 04:06 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:49096: write: broken pipe"}
fsb  | [GIN] 2024/02/14 - 16:06:00 | 206 |  360.049451ms |  155.94.xxx.xxx | GET      "/stream/12?hash=3400fd&d=true"
fsb  | 14/02/2024 04:06 PM	INFO	routes.Stream	Content-Range	{"start": 1522972448, "end": 2030538891, "fileSize": 2030538892}
fsb  | 14/02/2024 04:06 PM	ERROR	routes.Stream	Error while copying stream	{"error": "write tcp 172.18.0.2:8038->155.94.xxx.xxx:40266: write: broken pipe"}
